File: src/test/java/kitchenpos/domain/FakeProductRepository.java
package kitchenpos.domain;

import kitchenpos.products.domain.Product;
import kitchenpos.products.domain.ProductRepository;

import java.util.*;

public class FakeProductRepository implements ProductRepository {

    private final HashMap<UUID, Product> inMemory = new HashMap<>();

    @Override
    public List<Product> findAllByIdIn(List<UUID> ids) {
        return ids.stream()
                .map(inMemory::get)
                .toList();
    }

    @Override
    public Optional<Product> findById(UUID id) {
        return Optional.ofNullable(inMemory.get(id));
    }

    @Override
    public Product save(Product product) {
        inMemory.put(product.getId(), product);
        return product;
    }

    @Override
    public List<Product> findAll() {
        return new ArrayList<>(inMemory.values());
    }
}

File: src/test/kotlin/kitchenpos/tobe/domain/product/repository/FakeProductRepository.kt
package kitchenpos.tobe.domain.product.repository

import kitchenpos.tobe.product.domain.entity.ProductV2
import kitchenpos.tobe.product.domain.repository.ProductRepositoryV2
import java.util.*

class FakeProductRepository : ProductRepositoryV2 {
    private val products: MutableMap<UUID, ProductV2> = mutableMapOf()

    override fun save(productV2: ProductV2): ProductV2 {
        products[productV2.id] = productV2
        return productV2
    }

    override fun findProductById(id: UUID): ProductV2? {
        return products[id]
    }

    override fun findAll(): List<ProductV2> {
        return products.values.toList()
    }

    override fun findAllByIdIn(ids: List<UUID>): List<ProductV2> {
        return products.filterKeys { ids.contains(it) }.values.toList()
    }
}
